### PR TITLE
chore(conf): remove deprecated config options

### PR DESCRIPTION
**Pull request by jschmid1**
### Summary

Removing deprecated config options for good

### Full changelog

Removed the following config options

* upstream_keepalive
* nginx_http_upstream_keepalive
* nginx_http_upstream_keepalive_requests
* nginx_http_upstream_keepalive_timeout
* nginx_http_upstream_directives
* nginx_http_status_directives
* nginx_upstream_keepalive
* nginx_upstream_keepalive_requests
* nginx_upstream_keepalive_timeout
* client_max_body_size
* client_body_buffer_size
* cassandra_consistency
* router_update_frequency
* nginx_optimizations

Internal: https://konghq.atlassian.net/browse/FT-2537

### CODE REVIEW

File: kong.conf.default [hunk 1]
```diff
@@ -1328,6 +1328,7 @@
                                  # - `strict`: the router will be rebuilt
                                  #   synchronously, causing incoming requests to
                                  #   be delayed until the rebuild is finished.
+                                 #   (This is deprecated and will be removed soon)
```
Comment (flrgh) [inline kong.conf.default:1331]: IMO `soon` sounds a little vague. What do you think about `[...] will be removed in a future release` or something similar?

Comment (jschmid1) [reply kong.conf.default:1331]: done

**Comment by jschmid1** [comment]
These test failures are way too consistent to be unrelated.. Investigating

**Review by bungle — COMMENTED** [review]
I think we may need a changelog entry to this.

File: CHANGELOG.md [hunk 2]
```diff
@@ -452,6 +452,24 @@ In this release we continued our work on better performance:
...
+  - nginx_http_upstream_directives
+  - nginx_http_status_directives
+  - nginx_upstream_keepalive
+  - nginx_upstream_keepalive_requests
+  - nginx_upstream_keepalive_timeout
+  - client_max_body_size
+  - client_body_buffer_size
+  - cassandra_consistency
+  - router_update_frequency
+  - nginx_optimizations
```
Comment (kikito) [inline CHANGELOG.md:471]: We need to give more information to people considering updating to 3.0. I suggest something like this:

```suggestion
- Removed the following config options, which had been deprecated in previous versions, in favor of other config names. If you have any of these options in your config you will have to rename them: (removed option -> current option).
  - upstream_keepalive -> nginx_upstream_keepalive + nginx_http_upstream_keepalive
  - nginx_http_upstream_keepalive -> nginx_upstream_keepalive
  - nginx_http_upstream_keepalive_requests -> nginx_upstream_keepalive_requests
  - nginx_http_upstream_keepalive_timeout -> nginx_upstream_keepalive_timeout
  - nginx_http_upstream_directives -> nginx_upstream_directives 
  - nginx_http_status_directives -> nginx_status_directives
  - nginx_upstream_keepalive -> upstream_keepalive_pool_size
  - nginx_upstream_keepalive_requests -> upstream_keepalive_max_requests
  - nginx_upstream_keepalive_timeout -> upstream_keepalive_idle_timeout
  - client_max_body_size -> nginx_http_client_max_body_size
  - client_body_buffer_size -> nginx_http_client_max_buffer_size
  - cassandra_consistency -> cassandra_write_consistency / cassandra_read_consistency
  - router_update_frequency -> worker_state_update_frequency
- Removed the nginx_optimizations config option. If you have it in your configuration, please remove it before updating to 3.0.
```

**Review by kikito — APPROVED** [review]
Approved with one note about the changelog entry

**Review by flrgh — APPROVED** [review]

**Pull request merged**